src/reducers/vmDialog.js

```diff
@@ -0,0 +1,53 @@
+import Immutable from 'immutable'
+import { actionReducer } from './utils'
+
+const vmDialogReducer = actionReducer(Immutable.fromJS({
+  type: '',
+  vmId: '',
+  cluster: {},
+  template: {},
+  os: {},
+  name: '',
+  memory: '',
+  cpu: {},
+  errorMessage: '',
+}), {
+  UPDATE_CLUSTER (state, { payload: { cluster } }) {
+    const imUpdates = Immutable.fromJS(cluster)
```

Comment:
How about inlining those trivial intermediate variables? If after inlining it would appear too long you can also consider importing the `fromJS` function using a named import: `import { fromJS } from 'immutable'`.